tableauserverclient/server/endpoint/datasources_endpoint.py

```diff
@@ -348,41 +354,23 @@ def update_connections(
 
         Returns
         -------
-        list of str
+        Iterable of str
             The connection LUIDs that were updated.
         """
-        from xml.etree.ElementTree import Element, SubElement, tostring
 
         url = f"{self.baseurl}/{datasource_item.id}/connections"
         print("Method URL:", url)
 
-        ts_request = Element("tsRequest")
-
-        # <connectionLuids>
-        conn_luids_elem = SubElement(ts_request, "connectionLuids")
-        for luid in connection_luids:
-            SubElement(conn_luids_elem, "connectionLuid").text = luid
-
-        # <connection>
-        connection_elem = SubElement(ts_request, "connection")
-        connection_elem.set("authenticationType", authentication_type)
-
-        if username:
-            connection_elem.set("userName", username)
-
-        if password:
-            connection_elem.set("password", password)
-
-        if embed_password is not None:
-            connection_elem.set("embedPassword", str(embed_password).lower())
-
-        request_body = tostring(ts_request)
-
+        request_body = RequestFactory.Datasource.update_connections_req(
+            connection_luids=connection_luids,
+            authentication_type=authentication_type,
+            username=username,
+            password=password,
+            embed_password=embed_password,
+        )
         response = self.put_request(url, request_body)
 
-        logger.info(
-            f"Updated connections for datasource {datasource_item.id}: {', '.join(connection_luids)}"
-        )
+        logger.info(f"Updated connections for datasource {datasource_item.id}: {', '.join(connection_luids)}")
         return connection_luids
```

Comment:
As outlined in this PR, the function works with any `Iterable[str]`, not just a list. The only problem is if the user were to pass in an `Iterator`, the `Iterator` gets consumed by this function and returns an exhausted `Iterator`. If the connection ids are in the response, we can generate it from the response and have it work in the general case.